servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/BeforeFinallyHttpOperator.java

```diff
@@ -170,15 +170,12 @@ public void onSuccess(@Nullable final StreamingHttpResponse response) {
                 // Invoking a terminal method multiple times is not allowed by the RS spec, so we assume we have been
                 // cancelled.
                 assert state == TERMINATED;
-                if (discardEventsAfterCancel) {
-                    return;
+                // Since we've been cancelled we need to drain the request body.
+                toSource(response.payloadBody()).subscribe(CancelImmediatelySubscriber.INSTANCE);
```

Comment:
`response.payloadBody()` -> `response.messageBody()`? may avoid intermediate translations when not necessary.

iirc this is the code @idelpivnitskiy may have questions about? were there cases where we were intentionally not doing this (e.g. special cases in NettyHttp2Server or similar class for h2 vs h1).

Reply:
+1 for using `response.messageBody()`, good catch!

I'm agreed that we need to subscribe and cancel for `discardEventsAfterCancel` case, need to think if we should also do that for `!discardEventsAfterCancel` case or keep it as-is